adr/file_organization.md

```diff
@@ -0,0 +1,80 @@
+# Organize files for FDS clients
+
+- **Status:** accepted
+- **Deciders:** @akdetrick, @rey-wright, @ahmedhashim-cbi, @sfrieson-cbi, @jjniczCB, @kchen-cbi
+- **Date:** 2020.04.24
+
+---
+
+Technical Story: UIP-362
+
+## Context and Problem Statement
+
+We would like to generally flatten the directory structure in FDS as well as improving the
+import statements for client applications. This will be a breaking change for a major
+version, as clients will need to update imports once we make the change.
+
+## Decision Drivers
+
+* Remove unnecessary hierarchy in components dir
+* Simplify documentation
+* Adopt the unofficial CBI standard of organizing by component directories (each component
+  has its own directory).
+* Improve ergonomics and performance of import statements for client applications
+
+## Decision Outcome
+
+### Import statements
+Because individual imports for each component has a negative performance impact, we decided
+to use destructured imports and rely on a tree shaking strategy:
+
+```diff
+-import Flex from '@cbinsights/fds/lib/components/layout/Flex';
+-import FlexItem from '@cbinsights/fds/lib/components/layout/FlexItem';
+-import TextInput from '@cbinsights/fds/lib/components/forms/TextInput';
++import { Flex, FlexItem, TextInput } from '@cbinsights/fds/components';
+```
+
+The "root" file for each import corresponds to an FDS code module.
+The `lib` directory should be published as the root of the package.
+
+code module   | Import example
+------------- | --------------------------------------
+`dictionary`  | `import { COLOR_TIMBERWOLF, COLOR_PURPLE } from '@cbinsights/fds/dictionary';`
+`components`  | `import { Avatar, TextInput } from '@cbinsights/fds/components/';`
+`icons`       | `import { CloudIcon, BellIcon } from '@cbinsights/fds/icons';`
+`assets`      | Use path `@cbinsights/fds/assets/<file>` where applicable
+`base-styles` | not applicable; we only publish this CSS as part of the main styles file in `assets/`
+
+### `components` file structure
+
+Categorization directories will be removed.
+Each component will have its own directory under `components/`.
+Tests, stories, private components, and helpers will be colocated in component
+directories.
+
+```
+components/
+├── Foo/
```

Comment:
I know this is a little late, but it might be nice in the future to add a test directory and files. An example of how we generally do it in cbi-site

```
├── Foo/
│   ├── __tests__/
│   │   └─ Foo.spec.js
│   └─ Foo.jsx

Reply:
Not too late. Do you think this might be overkill though? Wondering the benefits of providing a folder specifically for tests (especially if it comes pass that we're only writing one test file for our FDS components)

Reply:
Definitely a benefit of alignment though.

Reply:
In the past I was generally against it, but I'm starting to come around. I think it generally makes things cleaner.
1) It makes it visually easier to find test files when they're all in one dir.
2) It makes it visually easier to find your component file when it's not collated with tests:
```
component-a.jsx
component-a.test.js
component-b.jsx
component-b.test.js
component-c.jsx
component-c.test.js
```
3) It's the Jest suggestion.

These aren't huge reasons. More opinion than anything. I've found easier since working at CBI than my previous job where we organized in the fashion of the example above.